packages/client/test/events.spec.ts

```diff
@@ -44,34 +65,67 @@ describe('Events', () => {
-          done(err);
-        }
+    describe('provider implements events', () => {
+      it('The provider defines a mechanism for signalling the occurrence of an event`PROVIDER_READY`', (done) => {
+        const client = API.getClient();
+        client.addHandler(ProviderEvents.Ready, () => {
+          try {
+            expect(API.providerMetadata.name).toBe(MOCK_EVENT_PROVIDER.metadata.name);
+            expect(MOCK_EVENT_PROVIDER.initialize).toHaveBeenCalled();
+            done();
+          } catch (err) {
+            done(err);
+          }
+        });
+        API.setProvider(MOCK_EVENT_PROVIDER);
+      });
+  
+      it('It defines a mechanism for signalling `PROVIDER_ERROR`', (done) => {
+        //make sure an error event is fired when initialize promise reject
+        const client = API.getClient();
+        client.addHandler(ProviderEvents.Error, () => {
+          try {
+            expect(API.providerMetadata.name).toBe(MOCK_ERROR_EVENT_PROVIDER.metadata.name);
+            expect(MOCK_ERROR_EVENT_PROVIDER.initialize).toHaveBeenCalled();
+            done();
+          } catch (err) {
+            done(err);
+          }
+        });
+        API.setProvider(MOCK_ERROR_EVENT_PROVIDER);
       });
-      API.setProvider(MOCK_PROVIDER);
     });
 
-    it('It defines a mechanism for signalling `PROVIDER_ERROR`', (done) => {
-      //make sure an error event is fired when initialize promise reject
-      ERROR_MOCK_PROVIDER.events?.addListener(ProviderEvents.Error, () => {
-        try {
-          expect(API.providerMetadata.name).toBe(ERROR_MOCK_PROVIDER.metadata.name);
-          expect(ERROR_MOCK_PROVIDER.initialize).toHaveBeenCalled();
-          done();
-        } catch (err) {
-          done(err);
-        }
+    describe('provider does not implement events', () => {
+      it('The provider defines a mechanism for signalling the occurrence of an event`PROVIDER_READY`', (done) => {
+        const client = API.getClient();
+        client.addHandler(ProviderEvents.Ready, () => {
+          try {
+            expect(API.providerMetadata.name).toBe(MOCK_NO_EVENT_PROVIDER.metadata.name);
+            expect(MOCK_NO_EVENT_PROVIDER.initialize).toHaveBeenCalled();
+            done();
+          } catch (err) {
+            done(err);
+          }
+        });
+        API.setProvider(MOCK_NO_EVENT_PROVIDER);
+      });
+  
+      it('It defines a mechanism for signalling `PROVIDER_ERROR`', (done) => {
+        //make sure an error event is fired when initialize promise reject
+        const client = API.getClient();
+        client.addHandler(ProviderEvents.Error, () => {
+          try {
+            expect(API.providerMetadata.name).toBe(MOCK_ERROR_NO_EVENT_PROVIDER.metadata.name);
+            expect(MOCK_ERROR_NO_EVENT_PROVIDER.initialize).toHaveBeenCalled();
+            done();
+          } catch (err) {
+            done(err);
+          }
+        });
+        API.setProvider(MOCK_ERROR_NO_EVENT_PROVIDER);
```

Comment:
@RealAnna These are the new tests I added. Similar to what was here before, except the mock provider has no `events` object. They failed before my last change.